lib/plugins/aws/invokeLocal/index.js

```diff
@@ -148,9 +164,166 @@ class AwsInvokeLocal {
 
     const configuredEnvVars = this.getConfiguredEnvVars();
 
-    _.merge(process.env, lambdaDefaultEnvVars, credentialEnvVars, configuredEnvVars);
+    const promises = _.entries(configuredEnvVars).map(([name, value]) => {
+      return this.resolveResource(value).then(output => {
+        configuredEnvVars[name] = output;
+      });
+    });
+
+    return BbPromise.all(promises).then(() => {
+      Object.assign(process.env, lambdaDefaultEnvVars, credentialEnvVars, configuredEnvVars);
+    });
+  }
+
+  loadExportsImpl(token, cfExports) {
```

Comment:
Let's name it `resolveExports`.

Reply:
The name change has been done.  This still has the Impl postfix, as I haven't found a way to make class level assignments to memoizee.

Reply:
> as I haven't found a way to make class level assignments to memoizee.

Following just works:

```javascript
const { memoize } = require('lodash');

let counter = 0;

class Foo {
  constructor() {
    this.bar = memoize(this.bar);
  }
  bar(elo) {
    return ++counter;
  }
}

const foo1 = new Foo();
const foo2 = new Foo();

console.log(1, 1, foo1.bar(1));
console.log(1, 2, foo1.bar(2));

console.log(2, 1, foo2.bar(1));
console.log(2, 2, foo2.bar(2));

console.log(1, 1, foo1.bar(1));
console.log(1, 2, foo1.bar(2));

console.log(2, 1, foo2.bar(1));
console.log(2, 2, foo2.bar(2));

```

What exactly issues are you approaching?